src/content/config.ts

```diff
@@ -0,0 +1,225 @@
+import { z, defineCollection } from "astro:content"
+
+const anyApiCollection = defineCollection({
+  type: "content",
+  schema: z
+    .object({
+      section: z.string(),
+      date: z.string(),
+      title: z.string(),
+      isMdx: z.boolean(),
+      permalink: z.string(),
+      whatsnext: z.record(z.string(), z.string()).optional(),
+      metadata: z
+        .object({
+          title: z.string(),
+          description: z.string(),
+          image: z.object({
+            0: z.string(),
+          }),
+        })
+        .optional(),
+    })
+    .strict(),
+})
```

Comment:
Good work on this, I would recommend extracting the base frontmatter to a zod object and  [extend](https://github.com/colinhacks/zod#extend) this base object

```suggestion
const baseFrontmatter = z
    .object({
      section: z.string(),
      date: z.string(),
      title: z.string(),
      isMdx: z.boolean(),
      permalink: z.string(),
      whatsnext: z.record(z.string(), z.string()).optional()
      })

const anyApiCollection = defineCollection({
  type: "content",
  schema: baseFrontmatter.extend({
      metadata: z
        .object({
          title: z.string(),
          description: z.string(),
          image: z.object({
            0: z.string(),
          }),
        })
        .optional(),
    })
    .strict(),
})
```

Reply:
I have simplified this even further and created BaseFrontmatter and BaseCollection. All the content collections are inferred from BaseCollection. 
However, it would be beneficial to determine which frontmatter parameters are required and which ones are optional. I have declared the parameters that are not present in all files as optional. 
@fmonper1, could you take a look and share your opinion?